# 1. Perkenalan

Halo, nama saya Fauzan Nurachman atau bisa dipanggil Ojan dari batch FTDS 003

## Problem Statement

Definisikan problem statement berdasarkan penjabaran masalah di bawah yang akan teman-teman jawab menggunakan SQL!

## Penjabaran Masalah

1. Tampilkan tabel sampai 5 baris pertama
2. Berasal dari agensi travel mana saja nasabah yang klaim asuransi?
3. Tipe agensi apa yang paling banyak yang klaim asuransinya diterima?
4. Destinasi negara mana yang paling banyak diterima klaim asuransinya?
5. Berapa nilai minimum, maksimum, dan rata-rata usia yang pengajuan klaim asuransinya diterima?
6. Jika agen asuransi hanya mengurusi klaim yang tujuan negaranya Singapura, Thailand, dan Malaysia, berapa total komisi yang didapat dari klaim yang diterima? (anggap komisi dalam USD)

## Jawaban

1. Tampilkan tabel sampai 5 baris pertama

In [2]:
from google.colab import auth
from google.cloud import bigquery
import pandas as pd

auth.authenticate_user()
print('Authenticated')

project_id = "hacktiv8-fauzan" #GUNAKAN GCP PROJECT-ID KALIAN MASING-MASING
client = bigquery.Client(project=project_id)

Authenticated


In [3]:
head_5 = client.query('''
Select * 
from rock-wonder-317907.h8_datasets.travel_insurance
limit 5
''').to_dataframe()

In [4]:
head_5

Agency    Agency_Type Distribution_Channel              Product_Name  Claim  \
0    EPX  Travel Agency               Online         Cancellation Plan  False   
1    EPX  Travel Agency               Online         Cancellation Plan  False   
2    EPX  Travel Agency               Online  1 way Comprehensive Plan  False   
3    EPX  Travel Agency               Online         Cancellation Plan  False   
4    EPX  Travel Agency               Online  2 way Comprehensive Plan  False   

   Duration Destination  Net_Sales  Commision__in_value_ Gender  Age  
0        75        FIJI       48.0                   0.0   None   36  
1        47        FIJI       38.0                   0.0   None   36  
2        47        FIJI       40.0                   0.0   None   36  
3        48        FIJI       80.0                   0.0   None   36  
4        26        FIJI      100.0                   0.0   None   36

2. Berasal dari agensi travel mana saja nasabah yang klaim asuransi?

In [5]:
Agency = client.query('''
SELECT DISTINCT(Agency) 
FROM (SELECT * FROM `rock-wonder-317907.h8_datasets.travel_insurance`
WHERE Agency_Type = 'Travel Agency')
''').to_dataframe()

In [6]:
Agency

Agency
0    EPX
1    CWT
2    KML
3    LWC
4    CBH
5    CSR
6    CCR
7    ADM
8    TTW
9    TST

Pada tabel diatas, dapat kita lihat ada 10 agensi travel yang klaim asuransi. Ada agensi EPX, CWT, KML, LWC, CBH, CSR, CCR, ADM, TTW, dan TST. Agensi travel tersebut dikategorikan berdasarkan kolom Agency_Type dengan spesifik 'Travel Agency'.

3. Tipe agensi apa yang paling banyak yang klaim asuransinya diterima?

In [7]:
Agency_Type = client.query('''
SELECT DISTINCT(Agency_Type), count(Claim) AS Total_Claim 
FROM `rock-wonder-317907.h8_datasets.travel_insurance`
WHERE Claim = True
GROUP BY Agency_Type
ORDER BY Total_Claim DESC
''').to_dataframe()

In [8]:
Agency_Type

Agency_Type  Total_Claim
0       Airlines          591
1  Travel Agency          336

Pada tabel diatas, dapat kita lihat tipe agensi apa yang paling banyak claim asuransinya diterima. Yang pertama ada tipe agensi Airlines dengan total claim yang diterima sebanyak 591, kedua ada tipe agensi Travel Agency dengan total claim yang diterima sebanyak 336.

4. Destinasi negara mana yang paling banyak diterima klaim asuransinya?

In [9]:
Destination = client.query('''
SELECT DISTINCT(Destination), count(Claim) AS Total_Claim 
FROM `rock-wonder-317907.h8_datasets.travel_insurance`
WHERE Claim = True
GROUP BY Destination
ORDER BY Total_Claim DESC
''').to_dataframe()

In [10]:
Destination

Destination  Total_Claim
0                          SINGAPORE          562
1                              CHINA           46
2                          AUSTRALIA           38
3                      UNITED STATES           37
4                           THAILAND           34
5                           MALAYSIA           23
6                     UNITED KINGDOM           16
7                          HONG KONG           15
8                          INDONESIA           15
9                 KOREA, REPUBLIC OF           15
10                             JAPAN           14
11                          VIET NAM           12
12                             ITALY            9
13                             INDIA            8
14                            FRANCE            8
15                       PHILIPPINES            8
16                            CANADA            7
17                       NETHERLANDS            6
18         TAIWAN, PROVINCE OF CHINA            6
19                             SPAIN            5
20                           GERMANY            5
21                       SWITZERLAND            5
22                      SOUTH AFRICA            5
23                       NEW ZEALAND            4
24                             NEPAL            3
25                           ICELAND            3
26                    CZECH REPUBLIC            2
27                 BRUNEI DARUSSALAM            2
28              UNITED ARAB EMIRATES            2
29                             MACAO            1
30                            GREECE            1
31                            ISRAEL            1
32                            NORWAY            1
33                            TURKEY            1
34                           FINLAND            1
35                           MYANMAR            1
36                          BULGARIA            1
37                          CAMBODIA            1
38                         ARGENTINA            1
39                        COSTA RICA            1
40  LAO PEOPLE'S DEMOCRATIC REPUBLIC            1

Pada tabel diatas, dapat kita lihat terdapat 41 negara yang klaimnya diterima. Kemudian kita akan lihat negara mana yang klaim diterimanya paling tinggi. Yang pertama ada negara Singapore dengan total klaim yang diterima sebanyak 562, kedua ada negara China dengan total klaim yang diterima sebanyak 46, ketiga ada negara Aurstralia dengan total klaim yang diterima sebanyak 38.

5. Berapa nilai minimum, maksimum, dan rata-rata usia yang pengajuan klaim asuransinya diterima?

In [11]:
Age = client.query('''
SELECT MIN(Age) as Min_Age, MAX(Age) AS Max_Age, AVG(Age) AS AVG_Age
FROM `rock-wonder-317907.h8_datasets.travel_insurance`
WHERE Claim = True
''').to_dataframe()

In [12]:
Age

Min_Age  Max_Age    AVG_Age
0       14      118  38.634304

Pada tabel diatas, dapat kita lihat minimum usia, maximum usia dan usia rata-rata yang klaim asuransinya diterima. Minimun usia yang klaim asuransinya diterima pada usia 14 tahun, kemudian maximum usia yang klaim asuransinya diterima pada usia 118 tahun, terakhir rata-rata usia yang klaimnya diterima berada pada usia 38 tahun. Umur 38 tahun memang sudah umur yang memasuki banyak penyakit, sehingga range umur tersebut banyak yang klaim asuransi.

6. Jika agen asuransi hanya mengurusi klaim yang tujuan negaranya Singapura, Thailand, dan Malaysia, berapa total komisi yang didapat dari klaim yang diterima? (anggap komisi dalam USD)

In [21]:
Commision = client.query('''
SELECT SUM(Commision__in_value_) as Commision_USD
FROM `rock-wonder-317907.h8_datasets.travel_insurance`
WHERE Claim = True AND Destination = 'SINGAPORE' OR Destination = 'THAILAND' OR Destination = 'MALAYSIA'
''').to_dataframe()

In [22]:
Commision

Commision_USD
0       85688.72

Dari tabel diatas, kita dapat mengetahui berapa komisi yang didapat dari klaim asuransi yang diterima dengan kondisi agen asuransu hanya mengurusi klaim yang tujuan negaranya Singapore, Thailand, dan Malaysia. Kita dapat lihat komisi yang didapat oleh agen asuransi yang hanya mengurusi negara Singapore, Thailand dan Malaysia sebesar 85,688 USD. Komisi tersebut sangat besar karena negara Singapore merupakan negara dengan peringkat 1 yang total klaimnya diterima.

## Poin Analisis

Dari penjabaran yang sudah kamu bahas (pertanyaan 2-6), apa kesimpulan yang kamu dapat sampaikan?

Kesimpulan yang saya dapat dari nomor 2 sampai 6 yaitu ini merupakan data agensi asuransi yang klaim asuransinya pada berbagai negara tujuan. Ada 10 Agensi dengan kategori Agensi Travel yaitu EPX, CWT, KML, LWC, CBH, CSR, CCR, ADM, TTW, dan TST. Ada 2 tipe agensi travel yang klaimnya paling banyak diterima yaitu Airlines dengan 591 klaim yang diterima dan Travel Agency dengan 336 klaim yang diterima. Ada 41 negara yang klaimnya diterima. Negara dengan total klaim yang paling banyak diterima adalah Singapore dengan total klaim yang diterima sebanyak 562, kedua ada negara China dengan total klaim yang diterima sebanyak 46, ketiga ada negara Aurstralia dengan total klaim yang diterima sebanyak 38. Minimun usia yang klaim asuransinya diterima pada usia 14 tahun, kemudian maximum usia yang klaim asuransinya diterima pada usia 118 tahun, terakhir rata-rata usia yang klaimnya diterima berada pada usia 38 tahun. Umur 38 tahun memang sudah umur yang memasuki banyak penyakit, sehingga range umur tersebut banyak yang klaim asuransi. Dan terakhir komisi dari agen asuransi itu sangatlah besar, jika kita sebagai agen asuransi dan mengurusi klaim dengan tujuan negaranya Singapore, Thailand, dan Malaysia akan mendapatkan komisi sebanyak 85,688 USD. Mengapa bisa banyak karena negara Singapore merupakan negara yang paling banyak klaimnya diterima, sehingga agen asuransi pun mendapatkan banyak komisi.

# TERIMA KASIH, MATUR SUWUN, HATUR NUHUN, GRACIAS